homeassistant/components/camera/rpi_camera.py

```diff
@@ -78,24 +77,17 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
             CONF_HORIZONTAL_FLIP: config.get(CONF_HORIZONTAL_FLIP),
             CONF_VERTICAL_FLIP: config.get(CONF_VERTICAL_FLIP),
             CONF_FILE_PATH: config.get(CONF_FILE_PATH,
-                                       os.path.join(os.path.dirname(__file__),
-                                                    'image.jpg'))
+                                       hass.config.path('image.jpg'))
```

Comment:
Can you rename it to `camera.rpi_camera.jpg` ? Makes it clear where the image is from.

Reply:
Actually, why do we even do this, wouldn't it make more sense to just use a temporary file when none is given? No reason to put it in the config dir.

Reply:
For creating temp files and dirs https://docs.python.org/3/library/tempfile.html

Reply:
CONF_FILE_PATH: config.get(CONF_FILE_PATH,
        tempfile.NamedTemporaryFile(suffix='.jpg', delete=False).name)

Are you talking about something like this?
I'm a bit unfamiliar with the Home Assistant code-base still, but wouldn't we run into issues with multiple images being saved as Home Assistant is restarted? Because we aren't deleting the used image.
Or is the default generated temp file path saved somehow? 
Otherwise, we would have to handle deletion of the temp file when Home Assistant is restarted.

Reply:
I wouldn't instantiate it as default param, because that way you would still create a temp file even when CONF_FILE_PATH is specified.

You can listen to EVENT_HOME_ASSISTANT_STOP to run code when Home Assistant stops.